### PR TITLE
uplink/piecestore: Check SN piece hash timestamp

### PR DESCRIPTION
**Pull request by ifraixedes**
What: Uplink verifies that the timestamp of upload result piece hash got from a storage node is not older than the amount of time that the Satellite accepts (currently 2 hours).

Why: Because if uplink doesn't verify it, it counts as a valid result upload piece and cuts down the long tail of potential valid ones.
When uplinks count the upload results with a too old timestamp as valid, it may send a number of upload results equal to the optimal threshold, then the Satellite filters out the ones with a too old timestamp and then get only a total number of valid ones less than the optimal threshold, causing to reject the uplink request with an error response and aborting completely the upload operation.

Please describe the tests: No additional test, the current test should pass. 
Please describe the performance impact: N/A

It should close the ticket https://storjlabs.atlassian.net/browse/V3-2474

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: uplink/piecestore/verification.go [hunk 1]
```diff
@@ -14,13 +15,17 @@ import (
 	"storj.io/storj/pkg/signing"
 )
 
+const pieceHashExpiration = 2 * time.Hour
```
Comment (navillasa) [inline uplink/piecestore/verification.go]: should this be configurable or okay as const?

**Comment by littleskunk** [comment]
Time window will be changed with https://github.com/storj/storj/pull/3255

File: uplink/piecestore/verification.go (hunk 1 above)
Comment (ifraixedes) [reply uplink/piecestore/verification.go]: It was and it still is a const in the satellite part, so I applied the same.

https://github.com/storj/storj/blob/master/satellite/metainfo/metainfo.go#L36

Which is used in https://github.com/storj/storj/blob/master/satellite/metainfo/validation.go#L378

**Comment by ifraixedes** [comment]
> Time window will be changed with #3255

@littleskunk I've seen the changes in that PR but I'm not sure in how that impacts to this PR considering that the links posted to @navillasa question https://github.com/storj/storj/pull/3246#discussion_r334444143

If something has to be changed with those changes merged by #3255 I would need some guidance.

**Comment by ifraixedes** [comment]
@littleskunk if there is anything to change regarding your comment, please open a new ticket with it.

**Pull request merged**